tests/evaluate/test_default_evaluator.py

```diff
@@ -1979,3 +1981,179 @@ def __call__(self, _df, _metrics):
+    assert "eval_results_table.json" in artifacts
+    logged_data = pd.DataFrame(**results.artifacts["eval_results_table"].content)
+    pd.testing.assert_frame_equal(logged_data, data.assign(outputs=["a", "b"]))
+    assert "exact_match" in results.metrics
+
+
+def test_evaluate_question_answering_without_targets():
+    with mlflow.start_run() as run:
+        model_info = mlflow.pyfunc.log_model(
+            artifact_path="model", python_model=language_model, input_example=["a", "b"]
+        )
+        data = pd.DataFrame({"question": ["a", "b"]})
+        results = mlflow.evaluate(
+            model_info.model_uri,
+            data,
+            model_type="question-answering",
+        )
+
+    client = mlflow.MlflowClient()
+    artifacts = [a.path for a in client.list_artifacts(run.info.run_id)]
+    assert "eval_results_table.json" in artifacts
+    logged_data = pd.DataFrame(**results.artifacts["eval_results_table"].content)
+    pd.testing.assert_frame_equal(logged_data, data.assign(outputs=["a", "b"]))
+    assert results.metrics == {}
+
+
+def test_evaluate_question_answering_fails_to_load_metric():
+    with mlflow.start_run() as run:
+        model_info = mlflow.pyfunc.log_model(
+            artifact_path="model", python_model=language_model, input_example=["a", "b"]
+        )
+        data = pd.DataFrame({"question": ["a", "b"], "answer": ["a", "b"]})
+
+        with mock.patch("evaluate.load", side_effect=ImportError("mocked error")) as mock_load:
+            results = mlflow.evaluate(
+                model_info.model_uri,
+                data,
+                targets="answer",
+                model_type="question-answering",
+            )
+            mock_load.assert_called_once_with("exact_match")
+
+    client = mlflow.MlflowClient()
+    artifacts = [a.path for a in client.list_artifacts(run.info.run_id)]
+    assert "eval_results_table.json" in artifacts
+    logged_data = pd.DataFrame(**results.artifacts["eval_results_table"].content)
+    pd.testing.assert_frame_equal(logged_data, data.assign(outputs=["a", "b"]))
+    assert results.metrics == {}
+
+
+def test_evaluate_text_summarization_with_targets():
+    with mlflow.start_run() as run:
+        model_info = mlflow.pyfunc.log_model(
+            artifact_path="model", python_model=language_model, input_example=["a", "b"]
+        )
+        data = pd.DataFrame({"text": ["a", "b"], "summary": ["a", "b"]})
+        results = mlflow.evaluate(
+            model_info.model_uri,
+            data,
+            targets="summary",
+            model_type="text-summarization",
+        )
+
+    client = mlflow.MlflowClient()
+    artifacts = [a.path for a in client.list_artifacts(run.info.run_id)]
+    assert "eval_results_table.json" in artifacts
+    logged_data = pd.DataFrame(**results.artifacts["eval_results_table"].content)
+    pd.testing.assert_frame_equal(logged_data, data.assign(outputs=["a", "b"]))
+    assert "rouge1" in results.metrics
```

Comment:
are rouge2, rougeL, and rougeSum also included?

Reply:
Added!